torchmetrics/audio/stoi.py

```diff
@@ -0,0 +1,118 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Callable, Optional
+
+from torch import Tensor, tensor
+
+from torchmetrics.functional.audio.stoi import stoi
+from torchmetrics.metric import Metric
+
+
+class STOI(Metric):
+    r"""STOI (Short Term Objective Intelligibility, see [2,3]), a wrapper for the pystoi package [1].
+
+    Intelligibility measure which is highly correlated with the intelligibility of degraded speech signals, e.g., due
+    to additive noise, single/multi-channel noise reduction, binary masking and vocoded speech as in CI simulations. The
+    STOI-measure is intrusive, i.e., a function of the clean and degraded speech signals. STOI may be a good alternative
+    to the speech intelligibility index (SII) or the speech transmission index (STI), when you are interested in the effect
+    of nonlinear processing to noisy speech, e.g., noise reduction, binary masking algorithms, on speech intelligibility.
+    Description taken from [Cees Taal's website](http://www.ceestaal.nl/code/).
+
```

Comment:
add a note similar to 
https://github.com/PyTorchLightning/metrics/blob/3613a21dfaed5257656b18879dc12973027a7caa/torchmetrics/image/fid.py#L143-L144